compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -631,7 +631,8 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
     val tparamsTxt = withEnclosingDef(constr) { tparamsText(tparams) }
     val primaryConstrs = if (constr.rhs.isEmpty) Nil else constr :: Nil
     val prefix: Text =
-      if (vparamss.isEmpty || primaryConstrs.nonEmpty) tparamsTxt
+      if (constr.symbol.owner.is(Module)) " "
```

Comment:
Shouldn't this be the empty string?

Reply:
To print `object Foo { .. }` instead of `object Foo{ ... }`.

Reply:
But in `puzzle.decompiled` I see two spaces after `object Test`

Reply:
Finally fixed this one detail